packages/console/src/pages/ApplicationDetails/ApplicationDetailsContent/ProtectedAppSettings/index.tsx

```diff
@@ -95,8 +96,32 @@ function ProtectedAppSettings({ data }: Props) {
             placeholder={t('application_details.application_name_placeholder')}
           />
         </FormField>
+        <FormField
+          isRequired
+          title="protected_app.form.url_field_label"
+          description={cond(
+            getFieldState('protectedAppMetadata.origin').isDirty &&
+              'protected_app.form.url_field_modification_notice'
+          )}
+          tip={<span className={styles.tip}>{t('application_details.origin_url_tip')}</span>}
+        >
+          <TextInput
+            {...register('protectedAppMetadata.origin', {
+              required: true,
+              validate: (value) =>
+                validateUriOrigin(value) || t('protected_app.form.errors.invalid_url'),
+            })}
+            error={
+              // eslint-disable-next-line @typescript-eslint/prefer-nullish-coalescing
+              errors.protectedAppMetadata?.origin?.message ||
+              (errors.protectedAppMetadata?.origin?.type === 'required' &&
```

Comment:
When will the `type` become not required?

Reply:
It's always required, but the error mesage might be about something else, like the origin is already in use and what not. Here I'm just checking if the error is about missing the required field, if yes then show the corresponding "url_required" message.